lib/DxilPIXPasses/DxilDbgValueToDbgDeclare.cpp

```diff
@@ -703,48 +730,43 @@ void DxilDbgValueToDbgDeclare::handleDbgValue(
   }
 
   const OffsetInBits InitialOffset = PackedOffsetFromVar;
-  llvm::IRBuilder<> B(DbgValue->getCalledFunction()->getContext());
-  auto* instruction = llvm::dyn_cast<llvm::Instruction>(V);
-  if (instruction != nullptr) {
-    instruction = instruction->getNextNode();
-    if (instruction != nullptr && !llvm::isa<TerminatorInst>(instruction)) {
-      // Drivers may crash if phi nodes aren't always at the top of a block,
-      // so we must skip over them before inserting instructions.
-      do {
-        instruction = instruction->getNextNode();
-      } while (instruction != nullptr && 
-          llvm::isa<llvm::PHINode>(instruction) &&
-          !llvm::isa<TerminatorInst>(instruction));
+  auto* insertPt = llvm::dyn_cast<llvm::Instruction>(V);
+  if (insertPt != nullptr && !llvm::isa<TerminatorInst>(insertPt)) {
+    insertPt = insertPt->getNextNode();
+    // Drivers may crash if phi nodes aren't always at the top of a block,
+    // so we must skip over them before inserting instructions.
+    while (llvm::isa<llvm::PHINode>(insertPt)) {
+      insertPt = insertPt->getNextNode();
```

Comment:
I don't know how that would crash. If insertPt is phi, it would enter this loop and and exit until it isn't, at which point insertPt would be a terminator. How would it crash?